index.js

```diff
@@ -77,7 +77,7 @@ class PQueue {
 
 		if (!this._isPaused && this.queue.size > 0) {
 			this.queue.dequeue()();
-		} else {
+		} else if (this.queue.size <= 0) {
```

Comment:
I think it would be clearer to do it like this:

```js
if (this.queue.size > 0) {
	if (!this._isPaused) {
		this.queue.dequeue()();
	}
} else {
	this._resolveEmpty();
	this._resolveEmpty = () => {};

	if (this._pendingCount === 0) {
		this._resolveIdle();
		this._resolveIdle = () => {};
	}
}
```